docs/topics/settings.rst

```diff
@@ -604,6 +604,26 @@ If you want to disable it set to 0.
 
     This feature needs Twisted >= 11.1.
 
+.. setting:: DOWNLOAD_ALLOW_DATALOSS
+
+DOWNLOAD_ALLOW_DATALOSS
+-----------------------
+
+Default: ``False``
+
+Whether or not to allow broken responses, where declared ``Content-Length``
+does not match content sent by the server. If ``False``, these responses
+trigger a ``ResponseFailed`` error. If ``True``, these responses are allowed
```

Comment:
Yes, ``ResponseFailed`` is handled by the retry middleware. And I think there are several reasons why a ``ResponseFailed`` can be raised, hence the ``_DataLoss`` explicit check. I don't know if it can be a legit scenario where a response declares ``Content-Length: X`` and suddenly the connection is closed after less than ``X`` bytes transmitted (i.e.: server crashed?). In that case a retrying the request looks like a correct behavior.